front/src/HOC/withLogger.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+
+const getDisplayName = SomeComponent => {
+  return (
+    SomeComponent.displayName || SomeComponent.name || 'Component'
+  );
+};
+
+export const withLogger = (params = {}) => {
+  const { output, lifecycles = [] } = params;
+  const logProps = lifecycles.findIndex(el => el === 'props') !== -1;
+  const logDidMount =
+    lifecycles.findIndex(el => el === 'didMount') !== -1;
+  const logShouldUpdate =
+    lifecycles.findIndex(el => el === 'shouldUpdate') !== -1;
+  const logDidUpdate =
+    lifecycles.findIndex(el => el === 'didUpdate') !== -1;
+  const logWillUnmount =
+    lifecycles.findIndex(el => el === 'willUnmount') !== -1;
+  const log = data => {
+    if (output) {
```

Comment:
Does out output function change with time? Can we somehow avoid this check on every invocation of `log` function?

Reply:
we can do something like:
let log;
if (output) {
 log = data => output(data);
} else {
 log = data => console.log(data);
}